exec/plugin-commands-script-runners/src/regexpCommand.ts

```diff
@@ -0,0 +1,15 @@
+export function tryBuildRegExpFromCommand (
+  command: string
+): RegExp | null {
+  if (command.length < 3) {
+    return null
+  }
+  if (command[0] !== '/' || command.lastIndexOf('/') < 1) {
+    return null
+  }
+  try {
+    return new RegExp(command.slice(0, command.lastIndexOf('/')).slice(1))
+  } catch {
+    return null
+  }
```

Comment:
Support `/x/imgy`.

```suggestion
  // https://github.com/stdlib-js/regexp-regexp/blob/6428051ac9ef7c9d03468b19bdbb1dc6fc2a5509/lib/regexp.js
  const isRegExpStr = /^\/((?:\\\/|[^\/])+)\/([imgy]*)$/.test(command)
  if (!isRegExpStr) {
    return null
  }
  
  // https://stackoverflow.com/a/874742/6596777
  const match = command.match(new RegExp('^/(.*?)/([imgy]*)$'))
  if (!match) {
    return null
  }
  
  try {
    return new RegExp(match[1], match[2])
  } catch {
    return null
  }
```

Ref:
* https://github.com/stdlib-js/regexp-regexp/blob/6428051ac9ef7c9d03468b19bdbb1dc6fc2a5509/lib/regexp.js
* https://stackoverflow.com/a/874742/6596777

Reply:
There are other flac chalactor to present flag https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp/flags other than `imy`, and these are defined in EcmaScript RegExp specification. I think it is hard to keep up with RegExp flag for ecmascript starndart so it is better to simply split wih `/`

Reply:
Good catch! I have modified the code:

```typescript
  // https://github.com/stdlib-js/regexp-regexp/blob/6428051ac9ef7c9d03468b19bdbb1dc6fc2a5509/lib/regexp.js
  const isRegExpStr = /^\/((?:\\\/|[^\/])+)\/([dgimuys]*)$/.test(command)
  if (!isRegExpStr) {
    return null
  }
  
  // https://stackoverflow.com/a/874742/6596777
  const match = command.match(new RegExp('^/(.*?)/([dgimuys]*)$'))
  if (!match) {
    return null
  }
  
  try {
    return new RegExp(match[1], match[2])
  } catch {
    return null
  }
```

 Result:
```js
console.log(/build:.*/dgimuys.test("build:web")) // true
```

Reply:
In what case would regex flag be useful? A regex is already more powerful than it needs to be for this simple task of selecting scripts.

Reply:
Actually, only the `i` flags are useful. But if flags are not supported, I think it's fine.
We need to discuss what the behavior would be if `flags` were not supported:
1. Ingore `flags` and print `warning`
2. Ignore `flags`
3. Report an error

Reply:
* If ignore `flags`:

```ts
// https://github.com/stdlib-js/regexp-regexp/blob/6428051ac9ef7c9d03468b19bdbb1dc6fc2a5509/lib/regexp.js
const reRegExp = new RegExp(/^\/((?:\\\/|[^\/])+)\/([dgimuys]*)$/)
const match = command.match(reRegExp)

if (!match) {
  return null
}

// if need print warning
if (match[2]) {
  logger.warn("not support flags and will ignore")
}

try {
  return new RegExp(match[1])
} catch {
  return null
}
```

* If support `flags`:

```ts
// https://github.com/stdlib-js/regexp-regexp/blob/6428051ac9ef7c9d03468b19bdbb1dc6fc2a5509/lib/regexp.js
const reRegExp = new RegExp(/^\/((?:\\\/|[^\/])+)\/([dgimuys]*)$/)
const match = command.match(reRegExp)

if (!match) {
  return null
}

try {
  return new RegExp(match[1], match[2])
} catch {
  return null
}
```

Reply:
I'd vote for either reporting an error or consider it as script name. Although the second one may be confusing

Reply:
I would like to vote for reporting an error or ignoring flag and warning it.

Reply:
I'd vote for ignoring the flags and warning it.

Reply:
I updated the `tryBuildRegExpFromCommand` to report the error if flag is specified in the passed script command [67f62a1](https://github.com/pnpm/pnpm/pull/5871/commits/67f62a1fab401a6657f877899a046d7d2eccb1e1) 👍